etcdserver/api/rafthttp/http.go

```diff
@@ -213,46 +218,49 @@ func (h *snapshotHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	dec := &messageDecoder{r: r.Body}
 	// let snapshots be very large since they can exceed 512MB for large installations
 	m, err := dec.decodeLimit(uint64(1 << 63))
+	from := types.ID(m.From).String()
```

Comment:
Hmm.. I like what you doing here but if `err` is nil then won't the `m.From` fail? May be better to define `from` variable here but set it's value under the following '`if'` loop. Thanks!